framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1079,6 +1084,43 @@ protected void printOrStoreMessage(
         Trees.instance(processingEnv).printMessage(kind, message, source, root);
     }
 
+    /**
+     * printOrStoreMessage method with an added stack trace argument. The stack trace is printed
+     * when the dumpOnErrors option is enabled.
+     *
+     * @param kind the kind of message to print
+     * @param message the message text
+     * @param source the souce code position of the diagnostic message
+     * @param root the compilation unit
+     * @param trace the stack trace where the checker encountered an error
+     */
+    protected void printOrStoreMessage(
```

Comment:
I don't see a call of this method in this class when the `dumpOnErrors` flag is provided.
Is the flag handled for a SourceChecker?

Reply:
The ```dumpOnErrors``` flag stores the stack trace when the Checker Framework starts to print/store the error messages. The stack trace is stored in BaseTypeChecker, because after the error messages are stored, the stack trace starts to unwind for compound checkers. This is why the printOrStoreMessage is not seen in the stack trace because it wasn't part of it when the stack trace was stored. Hence, the functions calls that print and log the errors are not visible in the stack trace, but the user can observe where the error occurred.

Reply:
This is a new message in `SourceChecker`, but I don't see a call to it in `SourceChecker`.
There should be a call in `SourceChecker` as well, in case somebody directly uses/extends it.

Reply:
I have added the method call in this function so the stack trace will be printed if someone directly calls it.